server/src/main/java/org/elasticsearch/index/codec/CodecService.java

```diff
@@ -32,14 +33,17 @@ public class CodecService {
     /** the raw unfiltered lucene default. useful for testing */
     public static final String LUCENE_DEFAULT_CODEC = "lucene_default";
 
-    public CodecService(@Nullable MapperService mapperService, BigArrays bigArrays) {
+    public CodecService(@Nullable MapperService mapperService, BigArrays bigArrays, IndexSettings indexSettings) {
```

Comment:
Maybe we could avoid increasing the number of arguments and retrieve settings from the MapperService, like we do for the bloom filter?